src/util/misc.ts

```diff
@@ -46,27 +46,26 @@ export function touch(paths: string[] | string) {
   });
 }
 
-export function getFullUrl(webUrlHost: string, url: string, baseUrl: string) {
-  if (typeof url !== 'string' || !url) {
-    throw new Error(`Expected url to be a string, got [${url}] instead`);
-  }
+export function getFullUrl(url: string, baseUrl: string) {
   const urlObject = urlParser.parse(url, false, true);
   if (!urlObject.protocol) {
-    const baseUrlObject = baseUrl ? urlParser.parse(baseUrl, false, true) : {} as UrlWithStringQuery;
-    urlObject.protocol = urlObject.protocol || baseUrlObject.protocol || 'http:';
-    urlObject.host = urlObject.host || baseUrlObject.host || webUrlHost;
-
-    /* Relative path */
-    if (urlObject.pathname && urlObject.pathname.indexOf('/') !== 0 && baseUrlObject.pathname) {
-      urlObject.pathname = `${pathParser.dirname(baseUrlObject.pathname)}/${urlObject.pathname}`;
-    }
-
+    mutateToFullUrl(urlObject, baseUrl);
     url = urlParser.format(urlObject);
   }
-
   return url;
 }
 
+export function mutateToFullUrl(urlObject: any, baseUrl: string){
+  const baseUrlObject = baseUrl ? urlParser.parse(baseUrl, false, true) : {} as UrlWithStringQuery;
```

Comment:
We should make a small test here to test `urlObject.protocol` and just return if already test.